token-erc-20/chaincode-java/src/main/java/org/hyperledger/fabric/samples/erc20/ERC20TokenContract.java

```diff
@@ -0,0 +1,444 @@
+      currentBalance = Long.parseLong(currentBalanceStr);
+    }
+    // Used safe math .
+    long updatedBalance = Math.addExact(currentBalance, amount);
+
+    stub.putStringState(balanceKey.toString(), String.valueOf(updatedBalance));
+    // Increase totalSupply
+    String totalSupplyStr = stub.getStringState(TOTAL_SUPPLY_KEY.getValue());
+    long totalSupply = 0;
+    if (!stringIsNullOrEmpty(totalSupplyStr)) {
+      totalSupply = Long.parseLong(totalSupplyStr);
+    }
+    // Used safe math .
+    totalSupply = Math.addExact(totalSupply, amount);
+    stub.putStringState(TOTAL_SUPPLY_KEY.getValue(), String.valueOf(totalSupply));
+    Transfer transferEvent = new Transfer("0x0", minter, amount);
+    stub.setEvent(TRANSFER_EVENT.getValue(), transferEvent.toJSONString().getBytes(UTF_8));
+  }
+
+  /**
+   * Burn redeems tokens the minter's account balance. This function triggers a Transfer event.
+   *
+   * @param ctx the transaction context
+   * @param amount amount of tokens to be burned
+   */
+  @Transaction(intent = Transaction.TYPE.SUBMIT)
+  public void Burn(final Context ctx, final long amount) {
+
+    // Check minter authorization - this sample assumes Org1 is the central banker with privilege to
+    // burn tokens
+
+    String clientMSPID = ctx.getClientIdentity().getMSPID();
+    ChaincodeStub stub = ctx.getStub();
+    if (!clientMSPID.equalsIgnoreCase(ContractConstants.MINTER_ORG_MSPID.getValue())) {
+      throw new ChaincodeException(
+          "Client is not authorized to burn tokens", UNAUTHERIZED_SENDER.toString());
+    }
+
+    String minter = ctx.getClientIdentity().getId();
+
+    if (amount <= 0) {
+      throw new ChaincodeException(
+          "Burn amount must be a positive integer", INVALID_AMOUNT.toString());
+    }
+
+    CompositeKey balanceKey = stub.createCompositeKey(BALANCE_PREFIX.getValue(), minter);
+    String currentBalanceStr = stub.getStringState(balanceKey.toString());
+    if (stringIsNullOrEmpty(currentBalanceStr)) {
+      throw new ChaincodeException("The balance does not exist", BALANCE_NOT_FOUND.toString());
+    }
+    long currentBalance = Long.parseLong(currentBalanceStr);
+    long updatedBalance = Math.subtractExact(currentBalance, amount);
+
+    stub.putStringState(balanceKey.toString(), String.valueOf(updatedBalance));
+
+    // Decrease totalSupply
+    String totalSupplyBytes = stub.getStringState(TOTAL_SUPPLY_KEY.getValue());
+    if (stringIsNullOrEmpty(totalSupplyBytes)) {
+      throw new ChaincodeException("TotalSupply does not exist", NOT_FOUND.toString());
+    }
+    long totalSupply = Math.subtractExact(Long.parseLong(totalSupplyBytes), amount);
+    stub.putStringState(TOTAL_SUPPLY_KEY.getValue(), String.valueOf(totalSupply));
+
+    // Emit the Transfer event
+    final Transfer transferEvent = new Transfer(minter, "0x0", amount);
+    stub.setEvent(TRANSFER_EVENT.getValue(), transferEvent.toJSONString().getBytes(UTF_8));
+  }

```

Comment:
As in `Mint`, please add a console or logger output before returning from the chaincode.  E.g.: 
```
	log.Printf("minter account %s balance updated from %d to %d", minter, currentBalance, updatedBalance)

```